src/api/QGCCorePlugin.cc

```diff
@@ -435,14 +435,18 @@ void QGCCorePlugin::instrumentValueAreaCreateDefaultSettings(const QString& defa
     }
 }
 
-QQmlApplicationEngine* QGCCorePlugin::createRootWindow(QObject *parent)
-{
-    QQmlApplicationEngine* pEngine = new QQmlApplicationEngine(parent);
-    pEngine->addImportPath("qrc:/qml");
-    pEngine->rootContext()->setContextProperty("joystickManager", qgcApp()->toolbox()->joystickManager());
-    pEngine->rootContext()->setContextProperty("debugMessageModel", AppMessages::getModel());
-    pEngine->load(QUrl(QStringLiteral("qrc:/qml/MainRootWindow.qml")));
-    return pEngine;
+QQmlApplicationEngine* QGCCorePlugin::createQmlApplicationEngine(QObject* parent)
+{
+    QQmlApplicationEngine* qmlEngine = new QQmlApplicationEngine(parent);
+    qmlEngine->addImportPath("qrc:/qml");
+    qmlEngine->rootContext()->setContextProperty("joystickManager", qgcApp()->toolbox()->joystickManager());
+    qmlEngine->rootContext()->setContextProperty("debugMessageModel", AppMessages::getModel());
+    return qmlEngine;
+}
+
+void QGCCorePlugin::createRootWindow(QQmlApplicationEngine* qmlEngine)
+{
+    qmlEngine->load(QUrl(QStringLiteral("qrc:/qml/MainRootWindow.qml")));
```

Comment:
Hmm, what if I want to change this URL like it was possible in previous example? Could You fix this please to allow custom MainWindow as well? I was good in previous version to specify our own MainWindow. Or it is still possible by overriding `createRootWindow()`? 

Ok, seems it is still possible as it is `virtual`, thanks!